### PR TITLE
Have a real controller handling cron requests

### PR DESCRIPTION
**Pull request by DeepDiver1975**
## Description
OC_Json is deprecated - we need to start cleaning up the mess.
This is one on many steps

## Related Issue
#ajax

## Motivation and Context
Clean code is better code

## How Has This Been Tested?
- manual

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: core/Controller/CronController.php [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+	public function run() {
+		// Exit if background jobs are disabled!
+		$appMode = $this->config->getAppValue('core', 'backgroundjobs_mode', 'ajax');
+		if ($appMode === 'none') {
+			return new JSONResponse(['data' => ['message' => 'Background jobs disabled!']]);
+		}
+
+		// We call cron.php from some website
+		if ($appMode === 'cron') {
+			// Cron is cron :-P
```
Comment (patrickjahns) [inline core/Controller/CronController.php]: remove unnecessary comment?

**Comment by mmattel** [comment]
Documentation relevant

**Comment by mmattel** [comment]
@DeepDiver1975 
Proposal: ``Run background jobs as cron`` --> ``Set background jobs to run as cron job``

**Comment by DeepDiver1975** [comment]
@PVince81 @patrickjahns please review once more - THX

**Comment by DeepDiver1975** [comment]
> Proposal: Run background jobs as cron --> Set background jobs to run as cron job

This command is executing the jobs. I think the current description is reasonable. Maybe replace "Run" with "Execute"?

@mmattel

**Comment by DeepDiver1975** [comment]
@PVince81 mind reviewing? THX

File: core/Command/System/Cron.php [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+		// We call ownCloud from the CLI (aka cron)
+		if ($appMode !== 'cron') {
+			$this->config->setAppValue('core', 'backgroundjobs_mode', 'cron');
+		}
+
+		$progress = new ProgressBar($output);
+
+		// We only ask for jobs for 14 minutes, because after 15 minutes the next
+		// system cron task should spawn.
+		$endTime = \time() + 14 * 60;
```
Comment (PVince81) [inline core/Command/System/Cron.php]: I guess no unit test was added for this old code ? else we'd need to use the time factory

**Review by PVince81 — APPROVED** [review]
Fine if codecov is happy :+1: 

Nice to have: more tests

File: core/Command/System/Cron.php (hunk 2 above)
Comment (DeepDiver1975) [reply core/Command/System/Cron.php]: This code is covered by tests - but I simply don't assert on this value.

**Comment by mmattel** [comment]
@DeepDiver1975 do you plan to backport?

**Comment by DeepDiver1975** [comment]
> @DeepDiver1975 do you plan to backport?

no - I see this oc11 only

**Comment by mmattel** [comment]
thanks,
@settermjd important comment above by @DeepDiver1975.
Means when a doc PR gets filed, it MUST NOT be backported, but available for oC11.
BTW: how to handle that, how can we identify those PR´s which are for oC11+ only, maybe a new tag in doc?

**Comment by phil-davis** [comment]
@PVince81 @DeepDiver1975 I labelled this v11 only-in-master
Is this still true? No backport?

**Comment by PVince81** [comment]
@DeepDiver1975 please confirm whether this is safe for backport

**Comment by phil-davis** [comment]
Backport `stable10` #31914 

Backport is there if you want it.

**Pull request merged**